database_cleanup/models/purge_tables.py

```diff
@@ -72,7 +72,9 @@ def purge(self):
                     )
 
             self.logger.info("Dropping table %s", line.name)
-            self.env.cr.execute("DROP TABLE %s", (IdentifierAdapter(line.name),))
+            self.env.cr.execute(
+                "DROP TABLE %s CASCADE", (IdentifierAdapter(line.name),)
```

Comment:
This looks dangerous and I would never knowingly run this on a production database. Do you think it would work to drop *all* constraints for all tables in the `line` loop in a first pass and then drop the tables in a second pass?

Reply:
You're right :) I tried your proposition and it works, at least in my case. Updating the PR to this less dangerous solution ;)

Reply:
What used to be the case is that constraints between tables that were going to be dropped were removed explicitely. That can be considered safe, because the related tables were going to be dropped anyway. What happens now in the suggested change is that constraints from a table that is not going to be dropped to a table that is going to be dropped are being dropped explicitely.
I would not consider this safe, because the existence of such a constraint is an indication that it is not safe to drop the table in the first place. It also breaks dropping tables that still (only) depend on each other.

Reply:
Considering that
* The current code in this PR is unsafe,
* My original suggestion is very much what is already implemented in the released version of the module,
* The issue occurs when dealing with a manually created `x_` model according to the traceback,
I would make an exception only when dealing with a manually created model (c.q. table) that starts with `x_`